effect/jvm/src/test/scala/scalaz/effect/PromiseSpec.scala

```diff
@@ -0,0 +1,90 @@
+package scalaz.effect
+
+import org.specs2.Specification
+import scalaz.data.Disjunction.-\/
+
+class PromiseSpec extends Specification with RTS {
+
+  def is = "PromiseSpec".title ^ s2"""
+        Make a promise and retrieve its value correctly after complete it with:
+          `complete` to complete that promise with a specified value. $e1
+          `done` to complete that promise with a completed result.    $e2
+
+        Make a promise and retrieve its fail value after complete it with:
+          `error` to fail that promise with a specified error.  $e3
+          `done` to complete that promise with a failed result. $e4
+
+        Given a completed promise `done` returns false and get should return the first completed value. $e5
+
+        Make a promise and retrieve its Throwable value after interruption calling:
+          `done` to complete that promise with a terminated result.              $e6
+          `interrupt` with a specified throwable and interrupt all other fibers. $e7
+
+     """
+
+  def e1 =
+    unsafePerformIO(
+      for {
+        p <- Promise.make[Void, Int]
+        f <- p.complete[Void](32).fork
+        s <- f.join
+        v <- p.get
```

Comment:
fork/join can be eliminated in this example, because it has the same effect as doing the `complete` in the main fiber. i.e. change to:

```scala
...
 unsafePerformIO(
      for {
        p <- Promise.make[Void, Int]
        v <- p.complete[Void](32)
...
```